### PR TITLE
stop play isapplicable and invariant holds tests

### PR DESCRIPTION
**Pull request by jaydenpiao**
### Description

 Implemented StopPlay isApplicable and invariantHolds tests. isApplicable() determines if the play should start running. invariantHolds() determines if the play should continue running.

### Testing Done

 Unit tests.

### Resolved Issues

 #2253 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/play/stop_play_test.cpp [hunk 1]
```diff
@@ -47,6 +47,27 @@ class StopPlayTest : public SimulatedPlayTestFixture
...
+    auto stop_play = StopPlay(play_config);
+
+    world.updateGameState(
+        ::TestUtil::createGameState(RefereeCommand::STOP, RefereeCommand::STOP));
+
+    ASSERT_TRUE(stop_play.isApplicable(world));
+    ASSERT_TRUE(stop_play.invariantHolds(world));
+
+    world.updateGameState(
+        ::TestUtil::createGameState(RefereeCommand::NORMAL_START, RefereeCommand::STOP));
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/stop_play_test.cpp:71]: can we also have a check for when game state is HALT?

**Review by jonathanlew — COMMENTED** [review]
Please rename the PR to remove your username

**Review by akhilveeraghanta — COMMENTED** [review]
nothing more to add beyond existing comment

**Review by jonathanlew — COMMENTED** [review]
please fix CI

**Comment by jonathanlew** [comment]
closed due to #2438 superseding it

**Pull request closed without merging**